File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
  const answer = [];

  for (let i = 1; i <= Math.sqrt(yellow); i++) {
    // 나누어 떨어지면
    if (yellow % i === 0) {
      const width = yellow / i;
      const height = i;

      const checkValue = width * 2 + height * 2 + 4;
      if (checkValue === brown) {
        answer.push(width + 2, height + 2);
        return answer;
      }
    }
  }
}

File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for i in range(1, int(yellow**0.5) + 1):
        width = yellow / i
        height = i
        
        result = width * 2 + height * 2 + 4
        if brown == result:
            return [width + 2, height + 2]
